core/src/main/java/org/elasticsearch/common/unit/ByteSizeUnit.java

```diff
@@ -225,6 +234,23 @@ public long toPB(long size) {
 
     static final long MAX = Long.MAX_VALUE;
 
+    private final byte id;
```

Comment:
nit: how about simply using the ordinal of the enum? we would have to be careful with the order if we add more units, which we are not likely to do. Also, in any case we would have to be careful with be comp anyways.